fdbserver/CommitProxyServer.actor.cpp

```diff
@@ -918,18 +996,24 @@ ACTOR Future<Void> getResolution(CommitBatchContext* self) {
 	state Future<std::unordered_map<EncryptCipherDomainId, Reference<BlobCipherKey>>> getCipherKeys;
 	if (pProxyCommitData->isEncryptionEnabled) {
 		static std::unordered_map<EncryptCipherDomainId, EncryptCipherDomainNameRef> defaultDomains = {
-			{ SYSTEM_KEYSPACE_ENCRYPT_DOMAIN_ID, FDB_DEFAULT_ENCRYPT_DOMAIN_NAME },
-			{ ENCRYPT_HEADER_DOMAIN_ID, FDB_DEFAULT_ENCRYPT_DOMAIN_NAME }
+			{ SYSTEM_KEYSPACE_ENCRYPT_DOMAIN_ID, FDB_SYSTEM_KEYSPACE_ENCRYPT_DOMAIN_NAME },
+			{ ENCRYPT_HEADER_DOMAIN_ID, FDB_ENCRYPT_HEADER_DOMAIN_NAME },
+			{ FDB_DEFAULT_ENCRYPT_DOMAIN_ID, FDB_DEFAULT_ENCRYPT_DOMAIN_NAME }
 		};
 		std::unordered_map<EncryptCipherDomainId, EncryptCipherDomainNameRef> encryptDomains = defaultDomains;
 		for (int t = 0; t < trs.size(); t++) {
 			TenantInfo const& tenantInfo = trs[t].tenantInfo;
 			int64_t tenantId = tenantInfo.tenantId;
 			Optional<TenantNameRef> const& tenantName = tenantInfo.name;
-			// TODO(yiwu): In raw access mode, use tenant prefix to figure out tenant id for user data
 			if (tenantId != TenantInfo::INVALID_TENANT) {
 				ASSERT(tenantName.present());
 				encryptDomains[tenantId] = tenantName.get();
+			} else {
+				for (auto m : trs[t].transaction.mutations) {
```

Comment:
If there is no tenant map, could we set the two default domains unconditionally and not iterate over the mutations at all?

Reply:
We could but the current code abstracts the decision in `getEncryptDetailsFromMutationRef` method, not sure if saving a loop gains us much.

Also, it would impact if num_mutations per transaction are too large to matter, if not, then would leave the code as is.

Reply:
The nice thing about extracting the check is that this avoids adding an unnecessary cost if you are using encryption in a non-tenant database. I don't really know the magnitude of the cost, and since this isn't our primary use-case maybe it's ok to leave it like this for now. If there is no benefit to running the check for every mutation, though, it seems like we might as well pull it out.

Reply:
I'm not against pulling the check to this level and for non-tenant (or cluster has no tenants so far) cluster it does mean check happens for every mutation and is avoidable. 

I will take care of it in my subsequent ClearRange support patch. 
Thanks